### DISCUSSION TITLE
Different way resources files are handle in `Analysis` as `datas` parameter

### DISCUSSION THREAD

**Discussion by emcek** [pyinstaller/pyinstaller#7596] [Help] [answered]
I created strip version of my project:
https://github.com/emcek/pyi

Note: You can run locally with:
1. git clone https://github.com/emcek/pyi.git
2. create local virtual environment inside repo, named as `venv`
3. pip install -U -r requirements.txt
4. python dcspy/run.py
5. pyinstaller dcspy.spec

I got several places where I'm using (with various files) i.e.:
```python
dcspy_ico = Path(__file__).resolve().with_name('dcspy.ico')
default_yaml = Path(__file__).resolve().with_name('config.yaml')
```

Take a look inside dcspy.spec:
```
    datas=[('venv/Lib/site-packages/customtkinter', 'customtkinter'),
           ('dcspy/dcspy.ico', '.'),
           ('dcspy/config.yaml', 'dcspy'),
           ('dcspy/falconded.ttf', 'dcspy'),
           ('dcspy/dcspy.png', 'dcspy'),
           ('dcspy/G13.png', 'dcspy'),
           ('dcspy/G19.png', 'dcspy')],
```
in order to run appliaction with pyinstaller icon file needs to be `.` but rest of resources in `dcspy`, but all are in the same folder on the same level.

Note: `customtkinter` needs some extra jsons,so include path where your `site-packages` is.

When I replace: `('dcspy/dcspy.ico', '.'),` with: `('dcspy/dcspy.ico', 'dcspy'),`

then I got:

![image](https://user-images.githubusercontent.com/475312/234369586-cae6835f-da1b-437d-a652-18fee54819f2.png)

**Comment by rokm** [comment, marked as answer]
That's because your entry-point script is located within the `dcspy` package; so if you run unfrozen under python, `__file__` in `run.py` resolves to `[...]\dcspy\run.py` and `__file__` in `tk_gui.py` similarly resolves to `[...]\dcspy\tk_gui.py`.

But in the frozen application, the entry-point script is treated as being in the top-level application directory; so `__file__` in `run.py` resolves to `[sys._MEIPASS]\run.py` while `__file__` in `tk_gui.py` still resolves to `[appdir]\dcspy\tk_gui.py`. 

The best way to overcome this limitation in PyInstaller's treatment of entry-point script is to simply have an entry point script outside of the package, or move all `run` logic to another module and have your entry-point script only import `run` from that module (i.e., minimal entry-point script that does no `__file__` resolution).

**Reply by rokm** [reply]
Or, you can detect the frozen application using `getattr(sys, 'frozen', False)` in `run.py` and adjust the icon path for the frozen case.

**Comment by emcek** [comment]
I think got it! I spent sometime to troubleshooting, [documentation](https://pyinstaller.org/en/stable/runtime-information.html#using-file) was not clear to me. 
I will test both solutions, see which fit me best.
I was a real problem, more as interesting observation.
